### ISSUE TITLE
annotation-tool

### ISSUE THREAD

**Issue by G-YY** [sergiomsilva/alpr-unconstrained#53]
Hi,I use the annotation-tool in your repo,but it did not work for me ,it always display "NO_LABEL",do you  konw how to solve it?Thank you1 @sergiomsilva

**Comment by Programmerwyl** [comment]
This is an annotation tool
If there is not label info in the .txt  corresponding to the image, NO LABEL will be displayed
This indicates that you need to enter the label information yourself
![image](https://user-images.githubusercontent.com/22095854/55527884-72740000-56cd-11e9-8767-6a212998274d.png)
You can press space, then enter the information, and then press space to confirm the information
After completion, pressing n will automatically generate the corresponding annotation information

**Comment by Programmerwyl** [comment]
For the label box of the license plate, it is also marked by the operation area of these keys, such as c, a, d, s
To be honest, it doesn't work very well！
Often collapse

**Comment by PhilipsKoshy** [comment]
I used the annotation tool with hundreds of images, using 5 images each time. Using the mouse, place the cursor at the top left corner of the license plate, press c, a; then move the mouse to the top right corner of the license plate, press "a", move the mouse to bottom right, press "a", move finally to bottom left, press "a" for the last time; then press "space" bar; now you can enter the license number; press "Space" bar again; then press "n". This is the sequence I followed.

**Comment by khawar512** [comment]
Anyone work on annotation tool?. The tool mentioned in paper isn't working well

**Comment by PhilipsKoshy** [comment]
I have used the annotation tool mentioned in this repo for about 400 images. But, each time I ran the annotation tool commandline, I just used 5 images. So, I ran the commandline 80 times (80 iterations * 5 images/iteration = 400).

**Comment by khawar512** [comment]
After training on 400 images. Do you obtain a reasonable accuracy ?

**Comment by PhilipsKoshy** [comment]
Out of the three stages (Vehicle Detection (VD) + License Plate Detection (LPD) + OCR), the training mentioned in this repo is only for the LPD stage. VD and OCR uses the pre-trained models. Now, to answer your question, yes, the training improves the LPD. But, I abandoned this project as I did not get good results with two line plates. My guess is the OCR training is not good enough for two line plates. It was able to recognize good-looking, simulated two-line plates, but it failed miserably with real two-line plates. I posted that question in this guthub and have not got any answers so far.

**Comment by laurentdebricon** [comment]
@PhilipsKoshy can you tell me which project works for you ?

**Comment by khawar512** [comment]
@PhilipsKoshy  Please email me. I am also working on the same project

**Comment by jackmleitch** [comment]
> I have used the annotation tool mentioned in this repo for about 400 images. But, each time I ran the annotation tool commandline, I just used 5 images. So, I ran the commandline 80 times (80 iterations * 5 images/iteration = 400).

When you call 5 images as a time, what do you use for the width and height arguments? Do you resize all the images to be the same size? @PhilipsKoshy

**Comment by thangdc94** [comment]
For labeling tool, I suggest using `https://github.com/openalpr/plate_tagger`. 
It's easy to use but after labeling you have to convert result to `alpr-unconstrained` format.

For two-line plates, I'm using some work-around. 
You can see my answer here https://github.com/sergiomsilva/alpr-unconstrained/issues/72#issuecomment-844953324

**Comment by jackmleitch** [comment]
> Out of the three stages (Vehicle Detection (VD) + License Plate Detection (LPD) + OCR), the training mentioned in this repo is only for the LPD stage. VD and OCR uses the pre-trained models. Now, to answer your question, yes, the training improves the LPD (I had to train the LPD, as mine is for a different geographic region than what the repo LPD model is trained for). But, I abandoned this project as I did not get good results with two line plates, possibly due to poor OCR. My guess is the OCR training is not good enough for two line plates. It was able to recognize good-looking, simulated two-line plates, but it failed miserably with real two-line plates. I posted that question in this github and have not got any answers so far #72. Still looking for a good OCR stage...

I'm struggling to get improvements to the LPD with more training data. Could you please provide details to how you fine-tuned the model? Thanks in advance! @PhilipsKoshy
